File: leetcode/subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    return getSubsets([...new Set(nums)]);
};

function getSubsets(set)  { 
    const result = [];
    const n = set.length; 
    for (let i = 0; i < (1<<n); i++) {
        const subset = [];
        for (let j = 0; j < n; j++)  {
            if ((i & (1 << j)) > 0) {
                subset.push(set[j]);
            }
        }
        result.push(subset);
    } 
    return result;
} 

File: leetcode/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        result = []
        
        for i in range(1 << n):
            subset = []
            for j in range(n):
                if i & (1 << j):
                    subset.append(nums[j])
            result.append(subset)
        
        return result
